### PR TITLE
Feature: Added new MASK-STRING-SKIP-ENDS transformation

### PR DESCRIPTION
**Pull request by nishant8887**
## Problem

Added a new MASK-STRING-SKIP-ENDS transformation.

## Proposed changes

If it fixes a bug or resolves a feature request, be sure to link to that issue._
New transformation added. The transformation enables user to mask a string except the start and end n-characters.

## Types of changes

What types of changes does your code introduce to PipelineWise?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] Description above provides context of the change
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Unit tests for changes (not needed for documentation changes)
- [ ] CI checks pass with my changes
- [x] Bumping version in `setup.py` is an individual PR and not mixed with feature or bugfix PRs
- [ ] Commit message/PR title starts with `[AP-NNNN]` (if applicable. AP-NNNN = JIRA ID)
- [ ] Branch name starts with `AP-NNN` (if applicable. AP-NNN = JIRA ID)
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions

### CODE REVIEW

**Comment by nishant8887** [comment]
Requires pipelinewise-transform-field PR:
https://github.com/transferwise/pipelinewise-transform-field/pull/28
https://github.com/transferwise/pipelinewise-transform-field/pull/29

**Comment by nishant8887** [comment]
@Samira-El @koszti It would be nice if you help me with the review for this PR.

**Comment by koszti** [comment]
can you please fix the linter problems reported at https://github.com/transferwise/pipelinewise/pull/798/checks?check_run_id=3630644119 ?

**Comment by nishant8887** [comment]
> can you please fix the linter problems reported at https://github.com/transferwise/pipelinewise/pull/798/checks?check_run_id=3630644119 ?

Done 👍

File: pipelinewise/fastsync/commons/transform_utils.py [hunk 1]
```diff
@@ -330,3 +350,36 @@ def __mask_date_to_sql(cls, column: str, sql_flavor: SQLFlavor) -> str:
...
+            sql_flavor: the sql flavor to use
+
+        Raises: NotImplementedError if mask-string-skip-ends is not implemented for the given sql flavor
+
+        Returns: sql string equivalent of the mask-string-skip-ends
+        """
+        skip_ends_n = int(transform_type.value[-1])
+
+        if sql_flavor == SQLFlavor.SNOWFLAKE:
+            trans = '{0} = CASE WHEN LENGTH({0}) > 2 * {1} THEN ' \
```
Comment (Samira-El) [inline pipelinewise/fastsync/commons/transform_utils.py:372]: I believe that this is not masking the entire value if it's not long enough, which was something we discussed and changed in the transform-field.

File: pipelinewise/fastsync/commons/transform_utils.py [hunk 2]
```diff
@@ -330,3 +350,36 @@ def __mask_date_to_sql(cls, column: str, sql_flavor: SQLFlavor) -> str:
...
+                    'CONCAT(SUBSTRING({0}, 1, {1}), REPEAT(\'*\', LENGTH({0})-(2 * {1})), ' \
+                    'SUBSTRING({0}, LENGTH({0})-{1}+1, {1})) ' \
+                    'ELSE {0} END'.format(column, skip_ends_n)
+        elif sql_flavor == SQLFlavor.POSTGRES:
+            trans = '{0} = CASE WHEN LENGTH({0}) > 2 * {1} THEN ' \
```
Comment (Samira-El) [inline pipelinewise/fastsync/commons/transform_utils.py:377]: Same here, I believe that this is not masking the entire value if it's not long enough, which was something we discussed and changed in the transform-field.

File: pipelinewise/fastsync/commons/transform_utils.py (hunk 1 above)
Comment (nishant8887) [reply pipelinewise/fastsync/commons/transform_utils.py:372]: Yup... changing that

**Comment by Samira-El** [comment]
Can you please add this transformation to the test pipelines [here](https://github.com/transferwise/pipelinewise/tree/master/tests/end_to_end/test-project). mainly the one where target is PG or Snowflake?

File: pipelinewise/fastsync/commons/transform_utils.py (hunk 1 above)
Comment (nishant8887) [reply pipelinewise/fastsync/commons/transform_utils.py:372]: @Samira-El Done with change. Thanks for catching that. Forgot to update the behaviour in this PR. 👍

File: pipelinewise/fastsync/commons/transform_utils.py (hunk 2 above)
Comment (nishant8887) [reply pipelinewise/fastsync/commons/transform_utils.py:377]: Done

**Comment by nishant8887** [comment]
> Can you please add this transformation to the test pipelines [here](https://github.com/transferwise/pipelinewise/tree/master/tests/end_to_end/test-project). mainly the one where target is PG or Snowflake?

Sure

**Comment by nishant8887** [comment]
@Samira-El I have added some data as well as the transformation to the test pipelines. Not sure though how the tests work. Do you verify data somewhere?

**Comment by nishant8887** [comment]
@Samira-El Could you please trigger the circleci build process? I have made the recommended changes.

**Comment by nishant8887** [comment]
@Samira-El  @koszti Could you please help me in triggering the next tests in circleci?

**Comment by Samira-El** [comment]
Hey @nishant8887 , the remaining tests don't run for PRs coming from forks.

**Comment by Samira-El** [comment]
Can you rebase your branch?

**Comment by Samira-El** [comment]
Can you rebase again? 🙈

**Comment by nishant8887** [comment]
@Samira-El Done with the rebase again :)

**Review by Samira-El — APPROVED** [review]
Thanks for this, LGTM!

**Pull request merged**